src/Shared/PropertyHelper/PropertyHelper.cs

```diff
@@ -151,6 +151,7 @@ public void SetValue(object instance, object? value)
     /// A cached array of all public properties of the specified type.
     /// </returns>
     [RequiresUnreferencedCode("This API is not trim safe.")]
+    [RequiresDynamicCode("This API is not trim safe.")]
```

Comment:
```suggestion
    [RequiresDynamicCode("This API requires dynamic code because it makes generic types which may be filled with ValueTypes.")]
```

`RequiresDynamicCode` != `not trim safe`. Instead, it means that this API may need to generate and JIT compile code, which is not allowed in Native AOT.

Reply:
Resolved by #45604